test/e2e/query_test.go

```diff
@@ -103,6 +104,40 @@ func sortResults(res model.Vector) {
 	})
 }
 
+func TestSidecarNotReady(t *testing.T) {
+	t.Parallel()
+
+	e, err := e2e.NewDockerEnvironment("e2e_test_query")
+	testutil.Ok(t, err)
+	t.Cleanup(e2ethanos.CleanScenario(t, e))
+
+	prom, sidecar, err := e2ethanos.NewPrometheusWithSidecar(e, "alone", defaultPromConfig("prom-alone", 0, "", ""), "", e2ethanos.DefaultPrometheusImage())
+	testutil.Ok(t, err)
+	testutil.Ok(t, e2e.StartAndWaitReady(prom, sidecar))
+	testutil.Ok(t, prom.Stop())
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Sidecar should not be ready - it cannot accept traffic if Prometheus is down.
+	testutil.Ok(t, runutil.Retry(1*time.Second, ctx.Done(), func() (rerr error) {
+		req, err := http.NewRequestWithContext(ctx, "GET", "http://"+sidecar.Endpoint("http")+"/-/ready", nil)
+		if err != nil {
+			return err
+		}
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return err
+		}
+		defer runutil.CloseWithErrCapture(&rerr, resp.Body, "closing resp body")
+
+		if resp.StatusCode == 200 {
+			return fmt.Errorf("got status code %d", resp.StatusCode)
```

Comment:
```suggestion
			return fmt.Errorf("sidecar should not be ready, got HTTP status code %d OK", resp.StatusCode)
```

Reply:
Why `OK`?

Reply:
Because that's what usually goes after `200` in the HTTP response https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200? (:

Reply:
> Because that's what usually goes after `200` in the HTTP response https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200? (:

yes, in case someone doesn't know what status code `200` signifies.

Reply:
Printing the status code and OK the same time seems redundant.

Reply:
So, can we merge this then @yeya24 ?